### ISSUE TITLE
"New value type (int) is not in annotated types" for float * int

### ISSUE THREAD

**Issue by donquixote** [kalessil/phpinspectionsea#882]
Hello Vladimir ,
I don't know what we should do with this one..

```php
function foo(float $x, float $y) {
  $x = $x * 1;  // -> inspection warning "New value type (int) is not in annotated types"
  $y = .0 * 1;  // -> NO inspection warning
}
```
Inspection warning: "New value type (int) is not in annotated types"
In PHP > Php Inspections (EA Extended) > Architecture > Callable parameter usage violates definition

From https://3v4l.org/TOuFl (with "eol versions"):
- float * int is always float in PHP 7.x.
- float * int can be int for PHP 5.x and hhvm.

I think we should either show the inspection warning on both instances, or never.

Of course even better would be to have a `number` type hint, which allows `float|int`. I think this was discussed in internals list, but I don't remember the details.

**Comment by donquixote** [comment]
Maybe add a note about this in the inspection description.

**Comment by kalessil** [comment]
Hi Andreas, this should be covered by #879 - it'll be in the next release (2-4 weeks)

**Comment by donquixote** [comment]
Thanks for quick response!

#879 does not contain detailed explanations.
What is the spec behavior with your changes?
Is float * int considered to be always float, or do you support the PHP 5.x situation where float * int can be int?

**Comment by kalessil** [comment]
There `float * int`, `float + int`, `float - int` is evaluated to float.  The 5.x case is not considered.
